Build tied to specific platform toolset on windows

When building for `-msvc` with the VC++ 2015 build tools, the build fails:
> C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.Cpp.Platform.targets(57,5): error MSB8020: The build tools for v120 (Platform Toolset = 'v120') cannot be found. To build using the v120 build tools, please install v120 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\diggs\Projects\rustup-rs\target\debug\build\lzma-sys-8de572d4e0ca373d\out\build\windows\liblzma.vcxproj]

The project should probably not be tied to a specific platform toolset version.

--- a/lzma-sys/build.rs
+++ b/lzma-sys/build.rs
@@ -29,8 +29,6 @@ fn main() {
                         .unwrap_or(String::new());
     if target.contains("msvc") {
         println!("cargo:rustc-link-lib=static=liblzma");
-        let mut msbuild = gcc::windows_registry::find(&target, "msbuild")
-                              .expect("needs msbuild installed");
         let build = dst.join("build");
         let _ = fs::remove_dir_all(&build);
         let _ = fs::remove_dir_all(dst.join("lib"));
@@ -43,9 +41,20 @@ fn main() {
             "Release"
         };
 
-        run(msbuild.current_dir(build.join("windows"))
-                   .arg("liblzma.vcxproj")
-                   .arg(&format!("/p:Configuration={}", profile)));
+        let mut build_succeeded = false;
+        for platform_toolset in &["v150", "v140", "v120", "v110"] {
+            let mut msbuild = gcc::windows_registry::find(&target, "msbuild")
+                    .expect("needs msbuild installed");
+            if try_run(msbuild.current_dir(build.join("windows"))
+                    .arg("liblzma.vcxproj")
+                    .arg(&format!("/p:Configuration={}", profile))
+                    .arg(&format!("/p:PlatformToolset={}", platform_toolset))) {
+                build_succeeded = true;
+                break;
+            }
+        }
+        assert!(build_succeeded);
+
         t!(fs::create_dir(dst.join("lib")));
         t!(fs::create_dir(dst.join("include")));
         let platform = if target.contains("x86_64") {"X64"} else {"Win32"};
@@ -135,9 +144,13 @@ fn main() {
     }
 }
 
-fn run(cmd: &mut Command) {
+fn try_run(cmd: &mut Command) -> bool {
     println!("running: {:?}", cmd);
-    assert!(t!(cmd.status()).success());
+    t!(cmd.status()).success()
+}
+
+fn run(cmd: &mut Command) {
+    assert!(try_run(cmd));
 }
 
 fn cp_r(src: &Path, dst: &Path) {

